tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp

```diff
@@ -96,20 +101,43 @@ operation::ProgramWithCallbacks MorehAdam::create_program(
     const Tensor& grad,
     const Tensor& exp_avg,
     const Tensor& exp_avg_sq,
-    float lr, float beta1, float beta2, float eps, float weight_decay, uint32_t step, bool amsgrad,
+    float lr,
+    float beta1,
+    float beta2,
+    float eps,
+    float weight_decay,
+    uint32_t step,
+    bool amsgrad,
     const std::optional<std::reference_wrapper<const Tensor>> max_exp_avg_sq,
     const MemoryConfig& mem_config) {
-
-    std::vector<std::variant<Tensor, char*>> outputs{nullptr, nullptr, nullptr, nullptr, nullptr};
-
-    operation::run(
-        MorehAdam{
-            .inplace = true,
-            .lr = lr, .beta1 = beta1, .beta2 = beta2, .eps = eps, .weight_decay = weight_decay, .step = step, .amsgrad = amsgrad,
-            .output_mem_config = mem_config},
+    std::vector<Tensor> dummy_output_tensors = {
+        Tensor(operation::get_workers_for_op_output({param, grad, exp_avg, exp_avg_sq}, {max_exp_avg_sq}))};
+
+    operation::launch_op(
+        [lr, beta1, beta2, eps, weight_decay, step, amsgrad, mem_config](
+            const std::vector<Tensor>& input_tensors,
+            const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+            const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
+            return operation::run(
+                MorehAdam{
+                    .inplace = true,
+                    .lr = lr,
+                    .beta1 = beta1,
+                    .beta2 = beta2,
+                    .eps = eps,
+                    .weight_decay = weight_decay,
+                    .step = step,
+                    .amsgrad = amsgrad,
+                    .output_mem_config = mem_config},
+                input_tensors,
+                optional_input_tensors,
+                optional_output_tensors);
+        },
         {param, grad, exp_avg, exp_avg_sq},
+        dummy_output_tensors,
         {max_exp_avg_sq});
 
+    std::vector<std::variant<Tensor, char*>> outputs{nullptr, nullptr, nullptr, nullptr, nullptr};
```

Comment:
Could you change return type of ```moreh_adam```?
```[[maybe_unused]] std::vector<std::variant<Tensor, char*>>``` -> ```std::vector<std::optional<Tensor>>```

like following link
https://github.com/tenstorrent/tt-metal/blob/5ed51dc4439e50a493a503bde5f9e59da35b55ca/tt_eager/tt_dnn/op_library/moreh_groupnorm/moreh_groupnorm_op.hpp#L71